src/chips/nrf51822/uart.rs

```diff
@@ -0,0 +1,145 @@
+#![feature(core_intrinsics)]
+
+use core::mem;
+
+//use hil::uart;
+//use nvic;
+//use chip;
+use core::intrinsics;
+
+struct Registers {
+    startrx: u32,
```

Comment:
Replace these field types with `VolatileCell<u32>` (not necessary for the reserved fields) and replace instances of `volatile_load(field)` and `volatile_store(field, val)` with `field.get()` and `field.set(val)`, respectively